neural_network/2_hidden_layers_neural_network.py

```diff
@@ -0,0 +1,203 @@
+import numpy
+
+
+class NeuralNetwork:
+    def __init__(self, input_array: numpy.array, output_array: numpy.array) -> None:
+        """
+        input_array : input values for training the neural network.
+        output_array : expected output values of the given inputs.
+        """
+        self.input = input_array
+        # Initial weights are assigned randomly where first argument is the number
+        # of nodes in previous layer and second argument is the
+        # number of nodes in the next layer.
+
+        # random initial weights for the input layer
+        # self.input.shape[1] is used to represent number of nodes in input layer
+        # first hidden layer consists of 4 nodes
+        self.weights1 = numpy.random.rand(self.input.shape[1], 4)
+
+        # random initial weights for the first hidden layer
+        # first hidden layer has 4 nodes
+        # second hidden layer has 3 nodes
+        self.weights2 = numpy.random.rand(4, 3)
+
+        # random initial weights for the second hidden layer
+        # second hidden layer has 3 nodes
+        # output layer has 1 node
+        self.weights3 = numpy.random.rand(3, 1)
+
+        self.y = output_array
+        self.output = numpy.zeros(output_array.shape)
+
+    def feedforward(self) -> None:
+        """
+        feedforward propagation using sigmoid activation function between layers
+        return the last layer of the neural network
+
+        >>> input_val =numpy.array(([0,0,0],[0,0,0],[0,0,0]),dtype=float)
+        >>> output_val=numpy.array(([0],[0],[0]),dtype=float)
+        >>> nn = NeuralNetwork(input_val,output_val)
+        >>> res = nn.feedforward()
+        >>> array_sum = numpy.sum(res)
+        >>> array_has_nan = numpy.isnan(array_sum)
+        >>> print(array_has_nan)
+        False
+        """
+        # layer1 is the layer connecting the input nodes with
+        # the first hidden layer nodes
+        self.layer1 = sigmoid(numpy.dot(self.input, self.weights1))
```

Comment:
Same as the comment for `weights1, weights2, ...`
Descriptive names for `layer1, layer2, ...`